flink-kubernetes-operator/src/main/java/org/apache/flink/kubernetes/operator/FlinkOperator.java

```diff
@@ -113,6 +113,9 @@ private void overrideOperatorConfigs(ConfigurationServiceOverrider overrider) {
         } else {
             LOG.info("Configuring operator with {} reconciliation threads.", parallelism);
             overrider.withConcurrentReconciliationThreads(parallelism);
+            // feat: separate executor service for workflow
+            // https://github.com/java-operator-sdk/java-operator-sdk/pull/1371
+            overrider.withExecutorService(Executors.newFixedThreadPool(parallelism));
```

Comment:
Hi @SteNicholas Could you please explain why we need this extra config here? Is it ok if we control it with the same config?

Reply:
cc @csviri

Reply:
normally only ` overrider.withConcurrentReconciliationThreads(parallelism);` if you don't want to replace the executor service with a different flavor (thus other than a fixed thread pool).

see: https://github.com/java-operator-sdk/java-operator-sdk/blob/d9e869c3bf5319163d8d884529f6b4a926bd2d85/operator-framework-core/src/main/java/io/javaoperatorsdk/operator/api/config/ConfigurationService.java#L127-L129

Reply:
@csviri @morhidi, this change comes from [FlinkOperatorTest.java#L54](https://github.com/apache/flink-kubernetes-operator/blob/main/flink-kubernetes-operator/src/test/java/org/apache/flink/kubernetes/operator/FlinkOperatorTest.java#L54), which get the `ExecutorService` with default thread number 10 and the `concurrentReconciliationThreads() ` returns the default thread number.

Reply:
Talked to @csviri offline, he'll clarify this part in a bit.

Reply:
yes, sorry, this seems to be an issue with the overrider, created a PR to fix it
https://github.com/java-operator-sdk/java-operator-sdk/pull/1484

pls for now override the service creation like above: `overrider.withExecutorService`

Reply:
@morhidi, like @csviri mentioned, the current way that overrides the service creation is correct.

Reply:
@csviri, could the fix be released in 3.2.2? Otherwise I will remove this line when bumping the version to 4.0.0. cc @morhidi

Reply:
@SteNicholas yes, will backport it to 3.x